src/main.cpp

```diff
@@ -4512,6 +4518,15 @@ void static ProcessGetData(CNode* pfrom, const Consensus::Params& consensusParam
                         // else
                             // no response
                     }
+                    else if (inv.type == MSG_CMPCT_BLOCK)
+                    {
+                        // Note that while we will respond to all MSG_CMPCT_BLOCK requests here,
```

Comment:
What is the reason for this? I think we should check early and dosban peers who getdata a CMPCT_BLOCK that was not recently inved.